document/core/appendix/embedding.rst

```diff
@@ -0,0 +1,526 @@
+Hence, these syntactic classes can also be interpreted as types.
+
+.. _embed-error:
+
+Failure of an interface operation is indicated by an auxiliary syntactic class:
+
+.. math::
+   \begin{array}{llll}
+   \production{(error)} & \error &::=& \ERROR \\
+   \end{array}
+
+In addition to the error conditions specified explicitly in this section, implementations may also return errors when specific :ref:`implementation limitations <impl>` are reached.
+
+
+.. note::
+   Errors are abstract and unspecific with this definition.
+   Implementations can refine it to carry suitable classifications and diagnostic messages.
+
+
+
+.. index:: allocation, store
+.. _embed-store:
+
+Store
+~~~~~
+
+.. _embed-init-store:
+
+:math:`\F{init\_store}() : \store`
+..................................
+
+* Return the empty :ref:`store <syntax-store>`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{init\_store}() &=& \{ \SFUNCS~\epsilon,~ \SMEMS~\epsilon,~ \STABLES~\epsilon,~ \SGLOBALS~\epsilon \} \\
+   \end{array}
+
+
+
+.. index:: module
+.. _embed-module:
+
+Modules
+~~~~~~~
+
+.. index:: binary format
+.. _embed-decode-module:
+
+:math:`\F{decode\_module}(\byte^\ast) : \module ~|~ \error`
+...........................................................
+
+* If there exists a derivation for the :ref:`byte <syntax-byte>` sequence :math:`\byte^\ast` as a :math:`\Bmodule` according to the :ref:`binary grammar for modules <binary-module>`, yielding a :ref:`module <syntax-module>` :math:`m`, then return :math:`m`.
+
+* Else, return :math:`\ERROR`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{decode\_module}(b^\ast) &=& m && (\iff \Bmodule \longrightarrow m{:}b^\ast) \\
+   \F{decode\_module}(b^\ast) &=& \ERROR && (\otherwise) \\
+   \end{array}
+
+
+.. index:: text format
+.. _embed-parse-module:
+
+:math:`\F{parse\_module}(\codepoint^\ast) : \module ~|~ \error`
+...............................................................
+
+* If there exists a derivation for the :ref:`source <syntax-source>` :math:`\codepoint^\ast` as a :math:`\Tmodule` according to the :ref:`text grammar for modules <text-module>`, yielding a :ref:`module <syntax-module>` :math:`m`, then return :math:`m`.
+
+* Else, return :math:`\ERROR`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{parse\_module}(c^\ast) &=& m && (\iff \Tmodule \longrightarrow m{:}c^\ast) \\
+   \F{parse\_module}(c^\ast) &=& \ERROR && (\otherwise) \\
+   \end{array}
+
+
+.. index:: validation
+.. _embed-validate-module:
+
+:math:`\F{validate\_module}(\module) : \error^?`
+................................................
+
+* If :math:`\module` is :ref:`valid <valid-module>`, then return nothing.
+
+* Else, return :math:`\ERROR`.
+
+.. math::
+   \begin{array}{lclll}
+   \F{validate\_module}(m) &=& \epsilon && (\iff {} \vdashmodule m : \externtype^\ast) \\
```

Comment:
According to https://webassembly.github.io/spec/appendix/index-rules.html, this should be `m : externtype* -> externtype*`, right?